### PR TITLE
fix(DB/SAI) Add spore burst with 15% trigger chance

### PR DESCRIPTION
**Pull request by ChrisCGalbraith**
## Changes Proposed:
This PR proposes changes to:
-  [ ] Core (units, players, creatures, game systems).
-  [ ] Scripts (bosses, spell scripts, creature scripts).
-  [X] Database (SAI, creatures, etc).

If your pull request promotes complex changes that require a detailed explanation, please describe them in detail specifying what your solution is and what is it meant to address.

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/16593

## SOURCE:

The changes have been validated through:
- [ ] Live research (checked on live servers, e.g Classic WotLK, Retail, etc.)
- [ ] Sniffs (remember to share them with the open source community!)
- [ ] Video evidence, knowledge databases or other public sources (e.g forums, Wowhead, etc.)
- [ ] The changes promoted by this pull request come partially or entirely from another project (cherry-pick). **Cherry-picks must be committed using the proper --author tag in order to be accepted, thus crediting the original authors, unless otherwise unable to be found**

## Tests Performed:

This PR has been:
- [X] Tested in-game by the author.
- [ ] Tested in-game by other community members/someone else other than the author/has been live on production servers.
- [ ] This pull request requires further testing and may have edge cases to be tested.

## How to Test the Changes:

- [x] This pull request can be tested by following the reproduction steps provided in the linked issue
- [ ] This pull request requires further testing. Provide steps to test your changes. If it requires any specific setup e.g multiple players please specify it as well.

1. Teleport to SSC - pull a sporebat, observe the spore burst being cast randomly (15%)

## Known Issues and TODO List:

Nothing TODO.

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: data/sql/updates/pending_db_world/rev_1691842029005931700.sql [hunk 1]
```diff
@@ -0,0 +1,6 @@
+UPDATE `creature_template` SET `AIName` = 'SmartAI' WHERE `entry` = 21246;
+
+DELETE FROM `smart_scripts` WHERE (`source_type` = 0 AND `entryorguid` = 21246);
+INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `event_param5`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_param4`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES
+(21246, 0, 0, 0, 0, 0, 100, 0, 2000, 12000, 20000, 25000, 0, 11, 39197, 0, 0, 0, 0, 0, 5, 25, 0, 0, 0, 0, 0, 0, 0, 'Serpentshrine Sporebat - In Combat - Cast Sonic Charge'),
+(21246, 0, 1, 0, 0, 0, 15, 0, 0, 0, 0, 0, 0, 11, 38471, 2, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 'Serpentshrine Sporebat - In Combat - Cast \'Spore Burst\'');
```
Comment (Kitzunu) [inline data/sql/updates/pending_db_world/rev_1691842029005931700.sql:6]: please add a newline at the end of the file

**Comment by Kitzunu** [comment]
What is the source for the timers?

**Comment by ChrisCGalbraith** [comment]
Got no sniffs for this.

 [Spore Burst](https://www.wowhead.com/tbc/spell=38471/spore-burst) indicates a 15% chance which is what I put in the `chance` field whilst leaving any timers blank.

**Comment by elthehablo** [comment]
> Got no sniffs for this.
> 
> [Spore Burst](https://www.wowhead.com/tbc/spell=38471/spore-burst) indicates a 15% chance which is what I put in the `chance` field whilst leaving any timers blank.

Why not make the event happen on aggro then? `event_type` 4

File: data/sql/updates/pending_db_world/rev_1691842029005931700.sql [hunk 2]
```diff
@@ -3,4 +3,4 @@ UPDATE `creature_template` SET `AIName` = 'SmartAI' WHERE `entry` = 21246;
 DELETE FROM `smart_scripts` WHERE (`source_type` = 0 AND `entryorguid` = 21246);
 INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `event_param5`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_param4`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES
 (21246, 0, 0, 0, 0, 0, 100, 0, 2000, 12000, 20000, 25000, 0, 11, 39197, 0, 0, 0, 0, 0, 5, 25, 0, 0, 0, 0, 0, 0, 0, 'Serpentshrine Sporebat - In Combat - Cast Sonic Charge'),
-(21246, 0, 1, 0, 0, 0, 15, 0, 0, 0, 0, 0, 0, 11, 38471, 2, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 'Serpentshrine Sporebat - In Combat - Cast \'Spore Burst\'');
\ No newline at end of file
+(21246, 0, 1, 0, 0, 0, 15, 0, 0, 0, 0, 0, 0, 11, 38471, 2, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 'Serpentshrine Sporebat - In Combat - Cast \'Spore Burst\'');
```
Comment (elthehablo) [inline data/sql/updates/pending_db_world/rev_1691842029005931700.sql:6]:
```sql
(21246, 0, 1, 0, 4, 0, 15, 0, 0, 0, 0, 0, 0, 11, 38471, 2, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 'Serpentshrine Sporebat - In Combat - Cast \'Spore Burst\'');

```

**Review by elthehablo — CHANGES_REQUESTED** [review]

**Review by elthehablo — CHANGES_REQUESTED** [review]
this makes it so it has a 15% chance of triggering on aggro, which I assume is the intended behaviour?

**Comment by Nyeriah** [comment]
On aggro is only cast once, on aggro, never repeats

**Comment by elthehablo** [comment]
> On aggro is only cast once, on aggro, never repeats

Right. Which I understood to be the correct behaviour needed here? Spell has a 15% cast to be triggered, on aggro?

**Comment by elthehablo** [comment]
> Why do you think that it is only ever cast on aggro?

Because you had no timers

**Comment by ChrisCGalbraith** [comment]
My understanding is that if no timers are implemented but a % chance is, then whenever the event is fired (which should be every combat update where no other event is fired) there is a 15% chance of that event running? 

Without actual sources though this entire thing is speculative.

**Comment by elthehablo** [comment]
> My understanding is that if no timers are implemented but a % chance is, then whenever the event is fired (which should be every combat update where no other event is fired) there is a 15% chance of that event running? 
> 
> Without actual sources though this entire thing is speculative.

But I think combat update is like 50ms so that would be 15% every 50ms essentially meaning it will be on average cast more than 2 times per second. Unless I am mistaken. Lemme take a look at smart scripts and how it deals with the parameters for IC event

**Comment by elthehablo** [comment]
Sorry on mobile. But I think this outlines that when the timers are set to 0 a no repeat flag is added. So I think this means that the event triggers once. But ofc this is logical when you have no repeat timer. This is why I suggested to change to SE 4

**Comment by ChrisCGalbraith** [comment]
Interesting, I see what you mean. but is this not specifically for `SMART_EVENT_TARGET_BUFFED`?.

I've just double checked on this branch and the spore bat IS repeating the event more than once in the same pull

**Comment by elthehablo** [comment]
> Interesting, I see what you mean. but is this not specifically for `SMART_EVENT_TARGET_BUFFED`?.
> 
> I've just double checked on this branch and the spore bat IS repeating the event more than once in the same pull

Okay I stand corrected

**Comment by elthehablo** [comment]
I think I was just very confused with how this is supposed to work hehe

**Review by Gultask — COMMENTED** [review]
OK!
So after some time I gathered what I could :p 

1. Charge ID is wrong. It should be 38461 with timers
Initial 4850-19400
Repeat 10900-14500

2. According to Mangos Spore Burst is
Initial 0-0
Repeat 24000-36800
with chance of 100

**Comment by Gultask** [comment]
> > My understanding is that if no timers are implemented but a % chance is, then whenever the event is fired (which should be every combat update where no other event is fired) there is a 15% chance of that event running?
> > Without actual sources though this entire thing is speculative.
> 
> But I think combat update is like 50ms so that would be 15% every 50ms essentially meaning it will be on average cast more than 2 times per second. Unless I am mistaken. Lemme take a look at smart scripts and how it deals with the parameters for IC event

If an event has a repeat timer of 0, it will automatically add the NOT_REPEATABLE flag during worldserver startup. Not only for TARGET_BUFFED since everything up in the case switches will also apply to them.

**Review by Kitzunu — APPROVED** [review]

**Pull request merged**